posts/tree-cell-typing/index.qmd

```diff
@@ -0,0 +1,241 @@
+
+```{r, eval = FALSE}
+plotProfileSimilarity(ioprofiles)
+```
+
+```{r, echo = FALSE}
+#| fig-cap: "Similarity matrix of reference profile cell types can be used to help inform cell type hierarchy."
+knitr::include_graphics("./figures/io_profile_similarity_heatmap.png")
+```
+
+As two examples from this heatmap, we can see that T cell subtypes as well as NK cells have high similarity and so should be grouped together. Additionally, endothelial and fibroblast cell types are highly similar, supporting our grouping of them into a 'structural' major cell type.
+
+Once we have this cell type tree, we can use `InSituTree` to run supervised cell typing with a nested approach.
+```{r runInSituTree, message=FALSE, eval = FALSE}
+# Load basic data consisting of:
+#   - counts: cell x gene count matrix
+#   - negmeans: vector of mean negative probe counts per cell
+
+load("mini_nsclc")
+
+# Run InSituTree with basic options
+res <- runInSituTree(
+  x = mini_nsclc$counts,
+  neg = rowMeans(mini_nsclc$neg),
+  full_profiles = ioprofiles,
+  cth = celltype_hierarchy_list,
+  cohort = NULL,
+  excluded_genes = NULL,
+  quantile_absolute_expression_difference_param = 0.5, # Include genes in the top 0.5 fraction in absolute expression differences
+  quantile_percent_expression_difference_param = 0.5, # Include genes in the top 0.5 fraction in percentage expression differences
+  return_summary_annotation = TRUE
+)
+
+# Access the final summarized cell typing results
+head(res$summaryAnnotation, 10)
+
+```
+
+```{r echo = FALSE, warning = FALSE, eval = TRUE}
+
+res <- readRDS("./assets/small_results.rds")
+
+head(res, 10)
+```
+
+# Additional recommendations {#sec-additional-recommendations}
+The structure of the cell type hierarchy list is a key determinant of how well this approach works. Major cell types that are unnecessarily broad may lead to poor cell typing if the average gene expression is too distinct from the expression of individual cells. As an example, the cell type lists available at our [CosMx Cell Profile library](https://github.com/Nanostring-Biostats/CosMx-Cell-Profiles/tree/main){target="_blank"} often contain an initial grouping of cell types into 'Tissue' or 'Immune'; we would recommend removing that large group and providing the next level down as top annotations. `InSituTree` does run quickly, so in cases of uncertainty we would recommend running the algorithm with multiple cell type hierarchies and choosing the one that gives the best cell typing. Evaluate the results as cell typing is typically evaluated: spatial distribution of cell types, top markers for each cell type, and clustering of similar cell types in UMAP space.
```

Comment:
It would be good to show one example to illustrate what you mean by "removing that large group and providing the next level down as top annotations." Besides, please put the second suggestion on running multiple cth for comparison in new paragraph. For the cell typing results evaluation, you can link to InSituType Q&A. https://github.com/Nanostring-Biostats/InSituType/blob/main/FAQs.md#interpreting-clustering-results

Reply:
I added a two column illustration of the full vs refined hierarchies. I also added the link and formatting suggestion.